lib/core/src/Cardano/Wallet/Api/Link.hs

```diff
@@ -676,15 +684,19 @@ getTransaction
         )
     => w
     -> t
+    -> Bool
```

Comment:
Hi @paolino 

Similarly to my previous comment, I feel this is introducing a bit of "boolean blindness".

If someone reading the integration tests for the first time sees `True` or `False` at call sites, they may wonder what the expected semantics are. To find out, they'd (presumably) navigate to the `Api.Link.getTransaction` function, but then be none the wiser, since the `Bool` value is simply passed to the inner function. They'd have to go all the way through the API, and then into the definition of `Api.Server.getTransaction` to figure out that:

- `True`
    maps to `TxMetadataNoSchema`
- `False`
    maps to `TxMetadataDetailedSchema`

This comprehension barrier might be compounded if we have multiple Boolean flags.

If it is necessary for callers of `Api.Link.getTransaction` to specify this flag, then I think we'd benefit from using the `TxMetadataSchema` type here. Readers of call sites would then be able to navigate to the definition of `TxMetadataSchema` and figure out the intended semantics by searching for uses of `TxMetadata` through the code base.